### DISCUSSION TITLE
ERROR while running Camry Impact Model in LS-DYNA® format

### DISCUSSION THREAD

**Discussion by panda1100** [OpenRadioss/OpenRadioss#418] [Q&A] [answered]
Hi All,

I'm trying to run "Camry Impact Model in LS-DYNA® format" Example model, the one introduced on examples page.
https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/10518559/Camry+Impact+Model+in+LS-DYNA+format

I got following error and couldn't finish this case. Does anyone can help me to solve this problem?

```
MESSAGE ID :        207
** RUN KILLED: TOTAL MASS ERROR LIMIT REACHED
NC=   15699 T= 1.0915E-02 DT= 2.3971E-10 ERR=-99.9% DM/M= 1.8663E+24
 ELAPSED TIME=       6941.96 s  REMAINING TIME=      69376.59 s
```

My procedures are following:

```
wget https://media.ccsa.gmu.edu/model/2012-toyota-camry-detailed-v5a.zip
unzip 2012-toyota-camry-detailed-v5a.zip
wget -O CamryOpenRadios.key https://openradioss.atlassian.net/wiki/download/attachments/10518559/CamryOpenRadioss.key?api=v2
mv CamryOpenRadios.key 2012-toyota-camry-detailed-v5a
cd 2012-toyota-camry-detailed-v5a

module load mpi/openmpi-x86_64
starter_linux64_gf -i CamryOpenRadios.key -np 8
mpirun -np 8 engine_linux64_gf_ompi -i CamryOpenRadios_0001.rad
```

Here attached is "CamryOpenRadios_0001.out" and "CamryOpenRadios_0001.rad"(I added .txt due to .out and .rad file cannot upload to discussion.)
[CamryOpenRadios_0001.out.txt](https://github.com/OpenRadioss/OpenRadioss/files/9539762/CamryOpenRadios_0001.out.txt)
[CamryOpenRadios_0001.rad.txt](https://github.com/OpenRadioss/OpenRadioss/files/9539765/CamryOpenRadios_0001.rad.txt)

**Comment by PaulAltair** [comment]
Thanks for reporting, I'm looking into this for you, the error is a typical one with a model instability, but the model was working ok in the last build I tried it with pre OpenRadioss release! The LS-DYNA input capability has been developing rapidly, so maybe something got changed/broken.

**Reply by PaulAltair** [reply]
I have traced this to a regression in treatment of foam material (DYNA MAT LAW57) in the bumper (vs latest commercial Altair Radioss release where the model runs fine) this is already being looked at. I'll let you know when it is working again!

**Reply by panda1100** [reply]
@PaulAltair Thank you very much for your support! I look forward to hearing from you. I'm planning to visualize data using ParaView (I'm waiting for converter :small_airplane:  ) after run this model.

**Reply by PaulAltair** [reply]
Great, this is early days! the ParaView converter is not far away I don't think. The foam model thing is just one of those things that gets broken during development unfortunately, OpenRadioss is being updated daily and sometimes changes can cause unintentional side effects!

**Reply by panda1100** [reply]
Yes, fun days as well! I failed with `latest-20220909` and currently testing with `latest-20220907`. I am using [Apptainer](https://github.com/apptainer/apptainer) container for this test that works really great with MPI app like OpenRadioss :smiley_cat: So far, I built latest-20220909, latest-20220908 and latest-20220907 container :+1:

**Reply by PaulAltair** [reply]
FYI I have proposed a temporary workaround for this for the time being (just to get you a running model) in the model page as a readme, if you substitute the *MAT_LOW_DENSITY_FOAM in bumper and Radiator(s) for *MAT_ELASTIC then it should run through
[README_September2022.txt](https://github.com/OpenRadioss/OpenRadioss/files/9543293/README_September2022.txt)

**Reply by panda1100** [reply]
@PaulAltair Thank you very much Paul! I'll try this workaround and will get back to you here :smiley:

**Reply by PaulAltair** [reply]
Pull requests have gone in today for the anim - vtk and T01 - csv converters so they should be available very soon

**Reply by panda1100** [reply]
@PaulAltair Your suggested workaround and VTK converter works great :rocket:
Thank you for your support!

**Reply by PaulAltair** [reply]
Great! I'll keep an eye on progress with the foam model too.

**Reply by panda1100** [reply]
Thank you very much!

**Reply by panda1100** [reply]
`th_to_csv` seems doesn't work at least for Camry example output.

I applied  [th_to_csv](https://github.com/OpenRadioss/OpenRadioss/pull/441) converter to output from ["Camry Impact Model in LS-DYNA® format" Example model](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/10518559/Camry+Impact+Model+in+LS-DYNA+format), and I got following error.

```
strace th_to_csv.linux64.exe CamryOpenRadiosT01
```

```
<I OMITTED LINES, TOO LONG TO PASTE>

read(3, "", 4096)                       = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x7ffcf2d43938} ---
+++ killed by SIGSEGV (core dumped) +++
Segmentation fault (core dumped)
```

**Reply by sebastienVilleneuve** [reply]
can you provide me your T01 file please , I'm not able to reproduce the problem

**Reply by panda1100** [reply]
Thank you, please find linked file.
https://drive.google.com/file/d/1biAm3lBW82XPTHOiDAsACpgIJ380D4dc/view?usp=sharing

(The file size exceeds limit of github discussion, I uploaded to google drive.)

**Reply by sebastienVilleneuve** [reply]
Hi, I can't reproduce your problem with the attached T01 file

**Reply by panda1100** [reply]
Thank you for your update! I actually pulled your code from your fork before it merged to main branch. I'll test with latest-20220913, and get back to you.

**Reply by sebastienVilleneuve** [reply]
![image](https://user-images.githubusercontent.com/100404030/190069369-726ee5a7-ac2b-478f-8acf-89de258e1a4f.png)

**Reply by sebastienVilleneuve** [reply]
Ok , this was the draft version , I cleaned some code in the 'final' version, certainly the reason of the issue you have .

Thanks a lot for your feedback, have a nice day .
Sébastien

**Reply by panda1100** [reply]
Thank you for taking your time. I tested with latest-20220913 but I still have the same issue.

cd /opt/OpenRadioss/tools/th_to_csv
gcc -g -o th_to_csv.linux64.exe th_to_csv.c

and

th_to_csv.linux64.exe CamryOpenRadiosT01

Is this the correct way of building code and usage?

**Reply by sebastienVilleneuve** [reply]
you can remove -g in the compilation.
and yes , ./th_to_csv.linux64.exe CamryOpenRadiosT01 is correct

**Reply by panda1100** [reply]
I removed -g but I still have the same issue. Hmm, what release did you use? main?

**Reply by panda1100** [reply]
I run my build with gdb and I got following error. I'm using data on google drive.  Is this help?

```
(gdb) next
92	    t01PreRead(t01Filename,&nbglobVar,&nbPartVar,&nbSubsVar,&nbTimeStep,&cptData,&cptThGroupNames);
(gdb) next
 ** ERROR: FILE (null) NOT FOUND
93	    ThPartNames =(char**)malloc(sizeof(char**)*nbPartVar);
```
and

```
(gdb) next
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) next
 ** ERROR: FILE (null) NOT FOUND
104	    csvFileWrite(csvFilename,titleFilename,&nbglobVar,&nbPartVar,&nbSubsVar,&nbTimeStep,&cptData,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) next

Program received signal SIGFPE, Arithmetic exception.
0x00000000004042fb in csvFileWrite (csvFilename=0x7fffffffd3e0 "(null).csv", titleFilename=0x7fffffffd300 "(null)_TITLES", nbglobVar=0x7fffffffd2f0, 
    nbPartVar=0x7fffffffd2ec, nbSubsVar=0x7fffffffd2e8, nbTimeStep=0x7fffffffd2fc, cptData=0x7fffffffd2f8, allData=0x7fffffffd2d0, ThPartNames=0x6088a0, 
    ThSubsNames=0x6088c0, ThGroupNames=0x6088e0) at th_to_csv.c:1139
1139	    int nbData = *cptData / *nbTimeStep ;
(gdb) next

Program terminated with signal SIGFPE, Arithmetic exception.
The program no longer exists.
```

I use `latest-20220913` and build/run procedures are following.

```
$ cd OpenRadioss-latest-20220913/tools/th_to_csv
$ gcc -g -o th_to_csv.linux64.exe th_to_csv.c
$ gdb ./th_to_csv.linux64.exe ~/Downloads/CamryOpenRadiosT01
$ (gdb) break main
$ (gdb) run
$ (gdb) next
```
Disk has plenty of space to play with and I checked owner and permission of working directory to be sure and it looks fine.

**Reply by SebVilleneuve** [reply]
Can you try to launch exe from T01 directory ?

**Reply by panda1100** [reply]
Yes. I tried and get exact same error. What release/branch and what OS (version) do you use for test?

**Reply by panda1100** [reply]
Update: I had mistake about gdb usage and this time I got proper(?) error.

```
gdb --args ./th_to_csv.linux64.exe ~/Downloads/CamryOpenRadiosT01
```

This gives me

```
(gdb) next
92	    t01PreRead(t01Filename,&nbglobVar,&nbPartVar,&nbSubsVar,&nbTimeStep,&cptData,&cptThGroupNames);
(gdb) next
93	    ThPartNames =(char**)malloc(sizeof(char**)*nbPartVar);
(gdb) next
94	    ThSubsNames =(char**)malloc(sizeof(char**)*nbSubsVar);
(gdb) next
95	    ThGroupNames =(char**)malloc(sizeof(char**)*cptThGroupNames);
(gdb) next
96	    float allData[cptData];
(gdb) next
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) next

Program received signal SIGSEGV, Segmentation fault.
0x0000000000400c98 in main (argc=2, argv=0x7fffffffd558) at th_to_csv.c:100
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) next

Program terminated with signal SIGSEGV, Segmentation fault.
The program no longer exists.
```

I did same test in T01 directory and got same error.

```
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) next

Program received signal SIGSEGV, Segmentation fault.
0x0000000000400c98 in main (argc=2, argv=0x7fffffffd5f8) at th_to_csv.c:100
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) next
```

```
(gdb) info locals
i = <optimized out>
csvFilename = "CamryOpenRadiosT01.csv\000\000\240\347\377\367\377\177", '\000' <repeats 18 times>, "\300\324\377\377\377\177\000\000\304\340\377\377\377\177\000\000\177\067\337\367\377\177", '\000' <repeats 18 times>, "\002\000\000\000\000\000\000\000\370\325\377\377"
t01Filename = "CamryOpenRadiosT01", '\000' <repeats 22 times>, "\240\347\377\367\377\177\000\000\000\000\000\000\000\000\000\000P\324\377\377\377\177\000\000`\324\377\377\377\177\000\000\070\353\377\367\377\177", '\000' <repeats 21 times>
titleFilename = "CamryOpenRadiosT01_TITLES\000\000\000\000\000\000\000@\003", '\000' <repeats 14 times>, "@\003\000\000@\003\000\000&\260be\000\000\000\000\370\324\377\377\377\177\000\000\060\325\377\377\377\177\000\000\070\353\377\367\377\177\000\000\240\347\377\367\377\177\000\000\347\a\271", <incomplete sequence \367>
ThPartNames = 0x6098b0
ThSubsNames = 0x61a940
ThGroupNames = 0x61a960
nbTimeStep = 1200
cptData = 11901601
cptThGroupNames = 1185
nbglobVar = 12
nbPartVar = 8720
nbSubsVar = 0
allData = <error reading variable allData (value requires 47606404 bytes, which is more than max-value-size)>
```
It looks like size of allData exceeds limit (max-value-size).

**Reply by sebastienVilleneuve** [reply]
can you unlimit your stacksize ?

**Reply by sebastienVilleneuve** [reply]
seems 'allData = <error reading variable allData (value requires 47606404 bytes, which is more than max-value-size)>' is a message from gdb.
could you launch without gdb ? puting exe & T01 in same dir .
I use CentOs machine with last th to csv code .

**Reply by panda1100** [reply]
Yes, I will change that value and get back to you, Thank you. I left my office just before I got your message, so get back to you tomorrow.

**Reply by sebastienVilleneuve** [reply]
thanks a lot , see you tomorrow

**Reply by panda1100** [reply]
I tried with `set max-value-size unlimited` and `puting exe & T01 in same dir`.
I'm testing `latest-20120914` on Rocky Linux 8.6 and GCC version is 8.5.0.


```
$ gdb --args ./th_to_csv.linux64.exe CamryOpenRadiosT01
$ (gdb) set max-value-size unlimited
$ (gdb) break main
$ (gdb) run
$ (gdb) next
```
For now I put log here. Around t01Read(), allData already free-ed before access??

```
Breakpoint 1, main (argc=2, argv=0x7fffffffd5f8) at th_to_csv.c:62
62	{
(gdb) next
71	    int nbTimeStep=0;
(gdb) 
72	    int cptData=0;
(gdb) 
73	    int cptThGroupNames=0;
(gdb) 
74	    int nbglobVar=0;
(gdb) 
75	    int nbPartVar=0;
(gdb) 
76	    int nbSubsVar=0;
(gdb) 
78	    if (argc == 2)
(gdb) 
80	        printf("\n T01 TO CSV CONVERTOR\n\n");
(gdb) 

 T01 TO CSV CONVERTOR

81	        printf("FILE    = %s\n", argv[1]);
(gdb) 
FILE    = CamryOpenRadiosT01
82	        printf("OUTPUT FILE    = %s.csv",argv[1]);
(gdb) 
83	        printf("\n");
(gdb) 
OUTPUT FILE    = CamryOpenRadiosT01.csv
86	    sprintf(csvFilename,"%s.csv",argv[1]);
(gdb) 
87	    sprintf(t01Filename,"%s",argv[1]);
(gdb) 
88	    sprintf(titleFilename,"%s_TITLES",argv[1]);
(gdb) 
92	    t01PreRead(t01Filename,&nbglobVar,&nbPartVar,&nbSubsVar,&nbTimeStep,&cptData,&cptThGroupNames);
(gdb) 
93	    ThPartNames =(char**)malloc(sizeof(char**)*nbPartVar);
(gdb) info locals
i = <optimized out>
csvFilename = "CamryOpenRadiosT01.csv\000\000\240\347\377\367\377\177", '\000' <repeats 18 times>, "\300\324\377\377\377\177\000\000\304\340\377\377\377\177\000\000\177\067\337\367\377\177", '\000' <repeats 18 times>, "\002\000\000\000\000\000\000\000\370\325\377\377"
t01Filename = "CamryOpenRadiosT01", '\000' <repeats 22 times>, "\240\347\377\367\377\177\000\000\000\000\000\000\000\000\000\000P\324\377\377\377\177\000\000`\324\377\377\377\177\000\000\070\353\377\367\377\177", '\000' <repeats 21 times>
titleFilename = "CamryOpenRadiosT01_TITLES\000\000\000\000\000\000\000@\003", '\000' <repeats 14 times>, "@\003\000\000@\003\000\000&\260be\000\000\000\000\370\324\377\377\377\177\000\000\060\325\377\377\377\177\000\000\070\353\377\367\377\177\000\000\240\347\377\367\377\177\000\000\347\a\271", <incomplete sequence \367>
ThPartNames = 0x405b4d <__libc_csu_init+77>
ThSubsNames = 0x0
ThGroupNames = 0x7ffff7ffe1f0
nbTimeStep = 1200
cptData = 11901601
cptThGroupNames = 1185
nbglobVar = 12
nbPartVar = 8720
nbSubsVar = 0
allData = 0x7fffffffd610
(gdb) next
94	    ThSubsNames =(char**)malloc(sizeof(char**)*nbSubsVar);
(gdb) next
95	    ThGroupNames =(char**)malloc(sizeof(char**)*cptThGroupNames);
(gdb) 
96	    float allData[cptData];
(gdb) info locals
i = <optimized out>
csvFilename = "CamryOpenRadiosT01.csv\000\000\240\347\377\367\377\177", '\000' <repeats 18 times>, "\300\324\377\377\377\177\000\000\304\340\377\377\377\177\000\000\177\067\337\367\377\177", '\000' <repeats 18 times>, "\002\000\000\000\000\000\000\000\370\325\377\377"
t01Filename = "CamryOpenRadiosT01", '\000' <repeats 22 times>, "\240\347\377\367\377\177\000\000\000\000\000\000\000\000\000\000P\324\377\377\377\177\000\000`\324\377\377\377\177\000\000\070\353\377\367\377\177", '\000' <repeats 21 times>
titleFilename = "CamryOpenRadiosT01_TITLES\000\000\000\000\000\000\000@\003", '\000' <repeats 14 times>, "@\003\000\000@\003\000\000&\260be\000\000\000\000\370\324\377\377\377\177\000\000\060\325\377\377\377\177\000\000\070\353\377\367\377\177\000\000\240\347\377\367\377\177\000\000\347\a\271", <incomplete sequence \367>
ThPartNames = 0x6098b0
ThSubsNames = 0x61a940
ThGroupNames = 0x61a960
nbTimeStep = 1200
cptData = 11901601
cptThGroupNames = 1185
nbglobVar = 12
nbPartVar = 8720
nbSubsVar = 0
allData = 0x7fffffffd610
(gdb) next
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) info locals
i = <optimized out>
csvFilename = "CamryOpenRadiosT01.csv\000\000\240\347\377\367\377\177", '\000' <repeats 18 times>, "\300\324\377\377\377\177\000\000\304\340\377\377\377\177\000\000\177\067\337\367\377\177", '\000' <repeats 18 times>, "\002\000\000\000\000\000\000\000\370\325\377\377"
t01Filename = "CamryOpenRadiosT01", '\000' <repeats 22 times>, "\240\347\377\367\377\177\000\000\000\000\000\000\000\000\000\000P\324\377\377\377\177\000\000`\324\377\377\377\177\000\000\070\353\377\367\377\177", '\000' <repeats 21 times>
titleFilename = "CamryOpenRadiosT01_TITLES\000\000\000\000\000\000\000@\003", '\000' <repeats 14 times>, "@\003\000\000@\003\000\000&\260be\000\000\000\000\370\324\377\377\377\177\000\000\060\325\377\377\377\177\000\000\070\353\377\367\377\177\000\000\240\347\377\367\377\177\000\000\347\a\271", <incomplete sequence \367>
ThPartNames = 0x6098b0
ThSubsNames = 0x61a940
ThGroupNames = 0x61a960
nbTimeStep = 1200
cptData = 11901601
cptThGroupNames = 1185
nbglobVar = 12
nbPartVar = 8720
nbSubsVar = 0
allData = <error reading variable allData (Cannot access memory at address 0x7ffffd2968b0)>
(gdb) next

Program received signal SIGSEGV, Segmentation fault.
0x0000000000400c98 in main (argc=2, argv=0x7fffffffd5f8) at th_to_csv.c:100
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
```

**Reply by panda1100** [reply]
I can not step into `t01Read`, somehow `t01Read` can not access to memory location of allData.

```
(gdb) step
96	    float allData[cptData];
(gdb) s
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) s

Program received signal SIGSEGV, Segmentation fault.
0x0000000000400c98 in main (argc=2, argv=0x7fffffffd5f8) at th_to_csv.c:100
100	    t01Read(t01Filename,allData,ThPartNames,ThSubsNames,ThGroupNames);
(gdb) s

Program terminated with signal SIGSEGV, Segmentation fault.
The program no longer exists.

```

**Reply by sebastienVilleneuve** [reply]
what is the stacksize on your machine ?

**Reply by panda1100** [reply]
**Thank you very much @sebastienVilleneuve!!** It works now. Sorry for late response.
I set stacksize to `unlimited`, that solved **MY** issue. Before that `ulimit -s` gave me `8192`.
I totally forgot about that. I didn't setup this fresh installation of Rocky as HPC purpose.

**Reply by sebastienVilleneuve** [reply]
my pleasure , You're welcome , Have a nice day.
Sébastien

**Reply by panda1100** [reply]
Have a nice day!
Yoshiaki

**Reply by PaulAltair** [reply]
With the latest build, the Low Density Foam-->Elastic material workaround is no longer required and the model should run as downloaded from CCSA with no modification

**Reply by panda1100** [reply]
@PaulAltair Thank you Paul for letting me know! That is a fantastic news!  I'll definitely try latest build this morning.

**Reply by panda1100** [reply]
@PaulAltair It is still running but it goes beyond `T= 1.0915E-02`. It looks good to me. Thank you @PaulAltair  and Thank you @sebastienVilleneuve for your support!

**Reply by panda1100** [reply]
@PaulAltair I couldn't finish with `latest-20220919`. I got following error. Does the latest means `latest-20220920`??

```
 -- RUPTURE OF SHELL ELEMENT :   3682408 AT TIME : 0.3103E-01
 NC=   44600 T= 3.1026E-02 DT= 7.4823E-08 ERR=  0.0% DM/M= 0.0000E+00
 ELAPSED TIME=      24047.86 s  REMAINING TIME=      68961.74 s
 -- RUPTURE OF SHELL ELEMENT :   3686720 AT TIME : 0.3103E-01

MESSAGE ID :        207

 ** RUN KILLED: TOTAL MASS ERROR LIMIT REACHED

 NC=   44686 T= 3.1030E-02 DT= 6.1066E-10 ERR= 99.9% DM/M= 4.7382E+23
 ELAPSED TIME=      24241.43 s  REMAINING TIME=      69505.13 s
     ANIMATION FILE: CamryOpenRadiosA008 WRITTEN
```

**Comment by PaulAltair** [comment]
Mine is still running well beyond that point, (I only have a virtual machine on my laptop so it is slow going), I built from main branch source code on Sunday: can you share your 0001.out, I can compare to mine

-- RUPTURE OF SHELL ELEMENT :   4333791 AT TIME : 0.4293E-01
 -- RUPTURE OF SHELL ELEMENT :   4346386 AT TIME : 0.4293E-01
 -- RUPTURE OF SHELL ELEMENT :   4467601 AT TIME : 0.4295E-01
 -- RUPTURE OF SHELL ELEMENT :   4478011 AT TIME : 0.4295E-01
 -- RUPTURE OF SHELL ELEMENT :   4334211 AT TIME : 0.4295E-01
 -- RUPTURE OF SHELL ELEMENT :   4346806 AT TIME : 0.4295E-01
 NC=   61700 T= 4.2979E-02 DT= 6.9669E-07 ERR= -3.7% DM/M= 8.4580E-03
 ELAPSED TIME=     112219.20 s  REMAINING TIME=     201102.26 s
 -- RUPTURE OF SHELL ELEMENT :   4467322 AT TIME : 0.4299E-01
 -- RUPTURE OF SHELL ELEMENT :   4477732 AT TIME : 0.4299E-01
 -- RUPTURE OF SHELL ELEMENT :   4467321 AT TIME : 0.4300E-01
 -- RUPTURE OF SHELL ELEMENT :   4477731 AT TIME : 0.4300E-01
 -- RUPTURE OF SHELL ELEMENT :   4467393 AT TIME : 0.4303E-01
 -- RUPTURE OF SHELL ELEMENT :   4477803 AT TIME : 0.4303E-01
 NC=   61800 T= 4.3049E-02 DT= 6.9669E-07 ERR= -3.7% DM/M= 8.4586E-03
 ELAPSED TIME=     112425.51 s  REMAINING TIME=     200963.99 s
 -- RUPTURE OF SHELL ELEMENT :   4334222 AT TIME : 0.4306E-01
 -- RUPTURE OF SHELL ELEMENT :   4346817 AT TIME : 0.4306E-01
 -- RUPTURE OF SHELL ELEMENT :   4334223 AT TIME : 0.4307E-01
 -- RUPTURE OF SHELL ELEMENT :   4346818 AT TIME : 0.4307E-01
 -- RUPTURE OF SHELL ELEMENT :   4467719 AT TIME : 0.4307E-01
 -- RUPTURE OF SHELL ELEMENT :   4478129 AT TIME : 0.4307E-01
 -- RUPTURE OF SHELL ELEMENT :   4333789 AT TIME : 0.4307E-01

**Reply by panda1100** [reply]
Thank you @PaulAltair . Please find attached file. [CamryOpenRadios_0001.out.txt](https://github.com/OpenRadioss/OpenRadioss/files/9605415/CamryOpenRadios_0001.out.txt)

**Reply by panda1100** [reply]
Here I attached is last 100 lines of `CamrOpenRadios_0000.out` [CamryOpenRadios_0000.out.trimed.txt](https://github.com/OpenRadioss/OpenRadioss/files/9605458/CamryOpenRadios_0000.out.trimed.txt)

**Reply by panda1100** [reply]
I'm testing against `latest-20220920` now. Attached files above are output of `latest-220919`.

**Comment by PaulAltair** [comment]
there is something way off there... the kinetic energy and mass are both way (way!) off what they should be, even at cycle 0. e.g. the mass reported for yours is 0.1283E+18 (units are tonnes, should be about 1.708!) are you modifying the launch script in any way? (you have changed its name?)

**Reply by panda1100** [reply]
Oh.. Here I attached is output of `latest-20220920` that I just started. I believe I didn't change starter file but I attached the file I'm using right now, just in case.

[CamryOpenRadios.key.txt](https://github.com/OpenRadioss/OpenRadioss/files/9605624/CamryOpenRadios.key.txt)
[CamryOpenRadios_0001.out.txt](https://github.com/OpenRadioss/OpenRadioss/files/9605625/CamryOpenRadios_0001.out.txt)

**Comment by PaulAltair** [comment]
hmm, yes, that is still wrong, but not by as much.... your initial mass is now 8.924 tonnes... my results are matching exactly the latest commercial mainline, so not sure what is happening in your case. I will try with latest OpenRadioss myself and get back to you

**Reply by PaulAltair** [reply]
downloaded source, recompiled and rerun, I still get the same values

**Reply by panda1100** [reply]
Could you please share your 0001.out? I will build main and would like to compare its output.

**Comment by PaulAltair** [comment]
[CamryOpenRadioss_0001.out.txt](https://github.com/OpenRadioss/OpenRadioss/files/9607350/CamryOpenRadioss_0001.out.txt)

That is mine so far, from main built this morning, hasn't run that far yet!

**Reply by panda1100** [reply]
Thank you Paul. I appreciate you taking the time for me.

**Reply by panda1100** [reply]
I used main ( `git show
commit ec64eee54bab8d011ebf024a0e052a43a6617be2`) but I couldn't replicate your 0001.out. I downloaded model from https://media.ccsa.gmu.edu/model/2012-toyota-camry-detailed-v5a.zip and starter file from https://openradioss.atlassian.net/wiki/download/attachments/10518559/CamryOpenRadioss.key?api=v2

**Comment by PaulAltair** [comment, marked as answer]
An Update on this: Thanks to this discussion a bug was found in the starter executable and corrected. 

Beyond that, with further testing on different machines, it seems this Camry Model is rather sensitive. The normal variations between results on different architectures in this case are enough to cause bifurcations in the run, often around the rh engine mount tube, this may cause a sudden jump in energy balance or even the model to exit prior to normal end time. With some experimentation, I have found that small modifications to the running timestep can help stability.

Because the model result is repeatable on any given machine (regardless of # of cores used), if the 'baseline' configuration of it works, there is no need to modify it, but if it fails on your set up, modification of the timestep should help.

I have updated the Camry model page with a note to this effect and a sample modified 'launch file'.

Thank you for your help identifying this issue.

**Reply by panda1100** [reply]
Thank you very much @PaulAltair for your generous support!!

**Comment by KameshENT** [comment]
![error](https://github.com/OpenRadioss/OpenRadioss/assets/164311258/b802613c-0d37-4f4d-97c0-a5b8043ef62c)
I'm getting the same error even after updating the CONTROL TIMESTEP card.
![image](https://github.com/OpenRadioss/OpenRadioss/assets/164311258/17950734-9b97-4cc1-87f4-77c9b1f51a19)
Kindly Help me out to solve this error..

Thanks

**Reply by MarianBulla** [reply]
Hi @KameshENT ,

it's hard to estimate the root cause for this, without knowing the model.
Usually, it is due to an unphysical load-case (e.g trying to push something through a rigid structure), or in the material propertied.

Can you provide your model, so I can double-check ?

Thanks and best regards,
Marian

**Reply by KameshENT** [reply]
I used the same model given in the OpenRadioss free examples in the following link
https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/10518559/Camry+Impact+Model+in+LS-DYNA+format

Thanks
Kamesh N

**Comment by WowCY2025** [comment]
Hi,

I encountered a similar issue with OpenRadioss. In my case, the calculation terminated due to a negative volume error.

I also tried running the same model with LS-DYNA, and the calculation was able to complete successfully. I have not yet performed a detailed comparison of the results or fully identified the root cause, but my current impression is that OpenRadioss may handle highly deformed solid elements somewhat differently from LS-DYNA.

It is possible that the issue is related to the element formulation, material/erosion settings, contact treatment, or the way the LS-DYNA input is interpreted by OpenRadioss. I will continue checking the model, but I wanted to share this observation in case it is related to the instability discussed here.

Best regards,

**Reply by WowCY2025** [reply]
I checked the animation results more carefully.

The elements reported by OpenRadioss, such as element 2024530 and 2024517, appear to be regular quadrilateral shell elements in the animation output. They are also located relatively far from the impact area, and I do not see any significant distortion of these elements before the error occurs.

The relevant message is:

```text
** WARNING ZERO OR NEGATIVE VOLUME: 3D-ELEMENT ID: 2024530
TOTAL STRAIN OPTION IS USED BY ELEMENT

FAILURE (FTAB) OF SHELL ELEMENT 2235841, GAUSS PT 1, LAYER 1, INTEGRATION PT 1 AT TIME: 4.6632E-02
FAILURE (FTAB) OF SHELL ELEMENT 3580237, GAUSS PT 1, LAYER 1, INTEGRATION PT 3 AT TIME: 4.6635E-02

WARNING: NODAL VELOCITY MAY BE TOO HIGH FOR INTERFACE 2000103
ERROR: NODAL VELOCITY IS TOO HIGH FOR INTERFACE 2000103
```

Based on the animation, I am not sure that elements 2024530 and 2024517 are the actual root cause of the termination. They do not show large deformation or obvious mesh distortion.

It may be more related to the shell failure reported just before the interface error, followed by an unstable contact/interface response or very high nodal velocity in interface 2000103.

The same model can run to completion in LS-DYNA, so I am wondering whether there may be a difference in the handling of local shell failure, contact release, or interface stability between OpenRadioss and LS-DYNA for this model.
